eox_tenant/signals.py

```diff
@@ -217,3 +225,97 @@ def clear_tenant(sender, request, **kwargs):  # pylint: disable=unused-argument
     Signal: django.core.signals.got_request_exception
     """
     pass
+
+
+def before_task_publish(sender, *args, **kwargs):  # pylint: disable=unused-argument
+    """
+    Dispatched before a celery task is published. Note that this is executed in the process sending the task.
+    See:
+        https://celery.readthedocs.io/en/latest/userguide/signals.html#before-task-publish
+    """
+    headers = kwargs.get('headers')
+    body = kwargs.get('body')
+    try:
+        request_info = body.get('args')[1].get('request_info')
+        headers['eox_tenant_sender'] = request_info.get('host') or None
+    except AttributeError:
+        # This is a async taks generated outside of the scope of a request i.e. CLI task
+        get_host_func = AsyncTaskHandler().get_host_from_task(sender)
+        headers['eox_tenant_sender'] = get_host_func(body)
+
+
+def task_prerun(sender, *args, **kwargs):  # pylint: disable=unused-argument
+    """
+    Dispatched before a task is executed.
+    See:
+       https://celery.readthedocs.io/en/latest/userguide/signals.html#task-prerun
+    """
+    headers = sender.request.get('headers', {})
+
+    http_host = headers.get('eox_tenant_sender', None)
+
+    if not http_host:
+        LOG.warning("Could not find the host information for eox_tenant.signals")
+
+    domain = http_host.split(':')[0]
+
+    can_keep = reset_or_keep_settings(domain)
+
+    if can_keep:
+        return
+
+    # Do the update
+    _update_settings(domain)
+
+
+class AsyncTaskHandler(object):
+    """
+    Handler used to get the tenant of an async task outside of the scope of a request.
+    """
+    TASKS = {
+        "openedx.core.djangoapps.schedules.tasks.ScheduleRecurringNudge": 'get_host_from_siteid'
```

Comment:
@morenol  why only this kind of task?

Reply:
This is the only CLI task that I know how to handle right now.